typhon/positioner.py

```diff
@@ -0,0 +1,190 @@
+import os.path
+import logging
+
+from ophyd import Device
+from qtpy import uic
+from qtpy.QtCore import Property, Slot
+
+from .plugins import register_signal
+from .utils import (TyphonBase, ui_dir, channel_from_signal, grab_kind,
+                    raise_to_operator, reload_widget_stylesheet)
+from .status import TyphonStatusThread
+from .widgets import TyphonDesignerMixin
+
+
+logger = logging.getLogger(__name__)
+
+
+class TyphonPositionerWidget(TyphonBase, TyphonDesignerMixin):
+    """
+    Widget to interact with an ``ophyd.Positioner``
+
+    Standard positioner motion requires a large amount of context for operator.
+    For most motors, in may not be enough to simply have a text field where
+    method can be punched in. Instead, information like soft limits and
+    hardware limit switches are crucial for a full understanding of the
+    position and behavior of a motor. This widget can supply a standard display
+    for all of these, but at a bare minimum the provided ``Device`` needs a
+    valid ``set`` function. The rest of the signals are searched for assuming
+    that the interface matches the ``EpicsMotor`` example supplied in
+    ``ophyd``.
```

Comment:
It'd be nice to reiterate those assumptions here.

Reply:
A good point! Added in d39367e